src/Jackalope/Security/Privilege.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Jackalope\Security;
+
+use PHPCR\Security\PrivilegeInterface;
+
+/**
+ * {@inheritDoc}
+ */
+class Privilege implements PrivilegeInterface
+{
+    /**
+     * @var $name
+     */
+    private $name;
+
+    /**
+     * @var PrivilegeInterface[]
+     */
+    private $declaredAggregate;
+
+    /**
+     * @var PrivilegeInterface[]
+     */
+    private $aggregate = array();
+
+    public function __construct($name, $declaredAggregate = array())
+    {
+        $this->name = $name;
+        $this->declaredAggregate = $declaredAggregate;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function isAbstract()
+    {
+        return false;
```

Comment:
afaik no, phpcr/jcr seems to have no way to make something abstract. ah, or its an implementation specific extension? then we would need to know how that fact is communicated by the server , and probably add a flag in the constructor.